packages/near-membrane-dom/src/browser-realm.ts

```diff
@@ -154,18 +219,8 @@ export default function createVirtualEnvironment(
     // We intentionally skip remapping Window.prototype because there is nothing
     // in it that needs to be remapped.
     env.lazyRemap(blueRefs.EventTargetProto, blueRefs.EventTargetProtoOwnKeys);
-    // We don't remap `blueRefs.WindowPropertiesProto` because it is "magical"
-    // in that it provides access to elements by id.
-    //
-    // Once we get the iframe info ready, and all mapped, we can proceed
-    // to detach the iframe only if the keepAlive option isn't true.
-    if (keepAlive) {
-        // TODO: Temporary hack to preserve the document reference in Firefox.
-        // https://bugzilla.mozilla.org/show_bug.cgi?id=543435
-        ReflectApply(DocumentProtoOpen, redDocument, []);
-        ReflectApply(DocumentProtoClose, redDocument, []);
-    } else {
-        ReflectApply(ElementProtoRemove, iframe, []);
+    if (realmController.needsExplicitCleanUp) {
```

Comment:
probably not need to expose `needsExplicitCleanUp`, but just always call cleanup, and let the controller implementation of the cleanup to take care if it.

Reply:
maybe rename cleanup to something more like `postInit`

Reply:
the issues with "Temporary hack to preserve the document reference in Firefox." can also be moved there to the postInit

Reply:
in fact, by looking at the code at the top, this is just not needed, the cleanup method is enough since for ShadowRealm it is a no-op anyways already.

Reply:
the realmController is not exported or exposed. It's all internal. This PR is not in a perf state. There'll be a follow-up to inline, optimize, lazy-ify things. Think of this as a barebones implementation or first draft.

Reply:
well, I reference to the `realmController` instance itself, very likely that will be moved to another file in the future, and most important, I believe this PR is missing an import part of the equation, which is about unforgeables that are not being redefined in the ShadowRealm global object because that code path needs to be forked as well. Probably that's the reason why this is failing. E.g.: this code today assumes that the realm's global `document` value is linked not remapped, but in the case of the shadow realm it is not defined at all, so it needs to be remapped. The controller will have to have that metainfo for the initialization to be carry on with one or the other.

Reply:
> well, I reference to the `realmController` instance itself, very likely that will be moved to another file in the future, 

My hunch is that `realmController` may not exist after a cleanup pass. 

>  I believe this PR is missing an import part of the equation, which is about unforgeables that are not being redefined in the ShadowRealm global object because that code path needs to be forked as well. Probably that's the reason why this is failing. E.g.: this code today assumes that the realm's global `document` value is linked not remapped, but in the case of the shadow realm it is not defined at all, so it needs to be remapped. The controller will have to have that metainfo for the initialization to be carry on with one or the other.

Good call.